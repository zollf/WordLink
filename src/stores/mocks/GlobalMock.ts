const GlobalMock: Global = {
  _ready: true,
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
  setUserInfo: jest.fn(),
  userInfo: {
    username: 'jeff',
    points: 20,
    puzzles: {
      completed: 2,
      easy: [],
      medium: [],
      hard: [],
    },
    hintsUsed: 6,
    hardestDifficulty: 'Medium',
    starsEarnt: 69,
  },
  init: jest.fn(),
};

export default GlobalMock;
