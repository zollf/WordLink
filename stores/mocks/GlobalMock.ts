const GlobalMock: Global = {
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
};

export default GlobalMock;
