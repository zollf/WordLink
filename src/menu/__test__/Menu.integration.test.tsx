import React from 'react';
import { Provider } from 'mobx-react';
import GlobalMock from '../../../stores/mocks/GlobalMock';

import Menu from '../';

import renderer from 'react-test-renderer';

const component = (
  <Provider global={GlobalMock}>
    <Menu />
  </Provider>
);

describe('<Menu />', () => {
  let wrapper = renderer.create(component);

  xit('difficulty select works correctly', () => {
    wrapper.root.findAllByProps({"data-test-id": "button"})[0].props.onPress();
    expect(GlobalMock.setCurrentDifficultyOpen).toBeCalled();
    expect(wrapper.toJSON().children[1].props['data-test-id']).toBeUndefined();
    wrapper = renderer.create(component);
    expect(wrapper.toJSON().children[1].props['data-test-id']).toBe('level-select');
  });

  xit('press all level buttons', () => {
    wrapper.root.findAllByProps({"data-test-id": "button"})[0].props.onPress();
    wrapper = renderer.create(component);

    wrapper.root.findAllByProps({"data-test-id": "button"})[2].props.onPress();
    expect(GlobalMock.setCurrentDifficultyOpen).toBeCalled();
    expect(wrapper.toJSON().children[1].props['data-test-id']).toBeUndefined();
    wrapper = renderer.create(component);
    expect(wrapper.toJSON().children[1].props['data-test-id']).toBe('level-select');

    wrapper.root.findAllByProps({"data-test-id": "button"})[0].props.onPress();
    wrapper = renderer.create(component);
    
    wrapper.root.findAllByProps({"data-test-id": "button"})[4].props.onPress();
    expect(GlobalMock.setCurrentDifficultyOpen).toBeCalled();
    expect(wrapper.toJSON().children[1].props['data-test-id']).toBeUndefined();
    wrapper = renderer.create(component);
    expect(wrapper.toJSON().children[1].props['data-test-id']).toBe('level-select');
  });
});