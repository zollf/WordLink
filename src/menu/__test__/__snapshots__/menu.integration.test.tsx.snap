// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
      "paddingBottom": 100,
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Raleway_400Regular",
        "fontSize": 24,
        "marginBottom": 24,
      }
    }
  >
    Level Select
  </Text>
  <View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 24,
          },
          Object {
            "opacity": mockConstructor {},
            "transform": Array [
              Object {
                "translateX": mockConstructor {},
              },
            ],
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4DCC4B",
            "borderRadius": 5,
            "display": "flex",
            "height": 76,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
            "width": 175,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Raleway_400Regular",
              "fontSize": 18,
            }
          }
        >
          EASY
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 24,
          },
          Object {
            "opacity": mockConstructor {},
            "transform": Array [
              Object {
                "translateX": mockConstructor {},
              },
            ],
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFBE40",
            "borderRadius": 5,
            "display": "flex",
            "height": 76,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
            "width": 175,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Raleway_400Regular",
              "fontSize": 18,
            }
          }
        >
          MEDIUM
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 24,
          },
          Object {
            "opacity": mockConstructor {},
            "transform": Array [
              Object {
                "translateX": mockConstructor {},
              },
            ],
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FF5740",
            "borderRadius": 5,
            "display": "flex",
            "height": 76,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
            "width": 175,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Raleway_400Regular",
              "fontSize": 18,
            }
          }
        >
          HARD
        </Text>
      </View>
    </View>
  </View>
  <View
    data-test-id="wrapper"
    style={
      Object {
        "alignItems": "center",
        "bottom": 80,
        "display": "flex",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      data-test-id="menu-bar"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#C8C8C8",
          "borderRadius": 5,
          "display": "flex",
          "flexDirection": "row",
          "height": 75,
          "justifyContent": "center",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 4,
          "width": 212,
          "zIndex": -1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "display": "flex",
            "height": 50,
            "justifyContent": "center",
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
            "width": 50,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../images/home.png",
            }
          }
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "display": "flex",
            "height": 50,
            "justifyContent": "center",
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 0.1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
            "width": 50,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../images/gamepad.png",
            }
          }
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "display": "flex",
            "height": 50,
            "justifyContent": "center",
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 0.1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
            "width": 50,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../images/avatar.png",
            }
          }
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F3F3F3",
              "borderRadius": 5,
              "height": 50,
              "position": "absolute",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 4,
              "width": 50,
              "zIndex": -1,
            },
            Object {
              "left": 10,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
